docs/guide/security-best-practices.md

```diff
@@ -185,6 +185,30 @@ class SiteController extends Controller
 }
 ```
 
+Disabling CSRF validation in [standalone actions](structure-controllers.md#standalone-actions) must be done in `init()`
+method. Do not place this code into `beforeRun()` method because it won't have effect.
+
+```php
+<?php
+
+namespace app\components;
+
+use yii\base\Action;
+
+class HelloWorldAction extends Action
+{
+    public function init()
+    {
+        parent::init();
+        \Yii::$app->controller->enableCsrfValidation = false;
+    }
+
+    public function run()
+    {
+        return "Hello World";
```

Comment:
It doesn't make sense to provide `run()` implementation like that. Either it should do something with POST or modify state or do something like that. Also instead of hello world I'd do delete action example and explicitly state that it's disabling CSRF is only when external website should be able to submit forms to your website. Also a warning like "Never do it if you do not plan external websites to ...".